content/js/users.md

```diff
@@ -1,40 +1,59 @@
 <a name="manage-user"></a>
 ## Manage User
 
+### How does it work
+
+Notice that even without logging in, user can still interact with the SDK,
+such as querying data in `skygear.publicDB`. However, this behavior can be
+changed by setting different access control for public. For more information,
+please refer to [access control](/js/guide/access-control#acl-default) page.
+Skygear uses `accessToken` with localstorage to track user authentication.
+Typical flow is shown below:
+
+1. Web page load/refresh
+2. SDK checks if user information exists in localstorage
+  - yes: SDK checks if `accessToken` has expired
+    + yes: user has to log in again
+    + no: user remains logged in (skip 3)
+  - no: user has to log in or sign up
+3. User logs in or signs up (trigger `onUserChanged`)
+4. SDK updates user information in container and localstorage
+  - `skygear.currentUser` is now available
+  - `skygear.accessToken` is now available
+5. User logs out or `accessToken` expires (trigger `onUserChanged`)
+6. SDK clears user information in container and localstorage
+
 ### Sign up
 
-If you are not familiar with Promises, please read more [here](https://www.promisejs.org/).
+To deal with duplicate username or email error, please refer to
+[handling errors](/js/guide/handling-errors#error-code) page.
 
 ``` javascript
-// We can signup using username or email. Both function will return a promise.
-let req = skygear.signupWithUsername(username, password);
-// Or by email.
-req = skygear.signupWithEmail(email, password);
+let req;
+req = skygear.signupWithUsername(username, password);
+req = skygear.signupWithEmail(email, password); // or by email
 
 req.then((user) => {
-  console.log('user id', user.id);
-  // only after the user successfully signup or login
-  // skygear.accessToken can be available
-  console.log('access token', skygear.accessToken);
+  console.log(skygear.currentUser);
+  console.log(skygear.accessToken);
 }, (error) => {
-  // maybe username or email already exists
-  // or the email format is incorrect
   console.error(error);
 });
 ```
 
 ### Log in
 
-Logging in is as straight-forward as signing up.
+To deal with nonexisting username or wrong password error, please refer to
+[handling errors](/js/guide/handling-errors#error-code) page.
 
 ``` javascript
-// If you signup with username, you can login by calling `loginWithUsername`.
-let req = skygear.loginWithUsername(username, password);
-// Or call `loginWithEmail` if you signup with email.
-req = skygear.loginWithEmail(username, password);
+let req;
+req = skygear.loginWithUsername(username, password);
+req = skygear.loginWithEmail(email, password); // or by email
```

Comment:
better separate the two lines (or simply describe the `loginWithEmail` in the paragraph. It is because this code snippet itself doesn't work, and is not what we will do.